app-backend/db/src/main/scala/ProjectLayerDao.scala

```diff
@@ -115,7 +115,14 @@ object ProjectLayerDao extends Dao[ProjectLayer] {
     query.filter(fr"project_id = ${projectId}").filter(layerId).selectOption
 
   def deleteProjectLayer(layerId: UUID): ConnectionIO[Int] =
-    query.filter(layerId).delete
+    for {
+      pl <- unsafeGetProjectLayerById(layerId)
+      _ = pl.overviewsLocation match {
```

Comment:
Since this is effectful, it makes sense to be truthful about it and wrap it in an IO -- you can do that in the function call and not have to wrap here